java/fury-core/src/main/java/org/apache/fury/BaseFury.java

```diff
@@ -64,9 +71,16 @@ public interface BaseFury {
    */
   <T> void registerSerializer(Class<T> type, Class<? extends Serializer> serializerClass);
 
+  void registerSerializer(Class<?> type, Serializer<?> serializer);
```

Comment:
The serializer is bind to a Fury instance, we may not be able to create a serializer object for registration unless we change the serializer interfere. Could we change this to register(Class type, Class serializer class)?

Reply:
There might be a singleton serializer. I've personally create some of them in my project. In this case they will be applied to each existing and each new Fury instances in `ThreadLocalFury` and `ThreadPoolFury`.
Maybe this is something that worth mentioning in the documentation.

However, if you didn't like it, we can remove this method and make user register singleton serializers using `execute` method. In this case, there is more chances that user will know what he is doing.

Reply:
I see, thanks. This is reasonable to add to BaseFury